### PR TITLE
Docs: Usage guidelines for SearchField, SegmentedControl and Tabs

### PR DESCRIPTION
**Pull request by ponori**
Added Usage guidelines for SearchField, SegmentedControl and Tabs

### CODE REVIEW

File: docs/pages/searchfield.js [hunk 1]
```diff
@@ -130,6 +130,30 @@ card(
...
 
+card(
+  <MainSection name="Usage guidelines">
+    <MainSection.Subsection columns={2}>
+      <MainSection.Card
+        cardSize="md"
+        type="do"
+        title="When to Use"
+        description={`
+          - To search or filter content within a surface..
```
Comment (ayeshakmaz) [inline docs/pages/searchfield.js:141]: double period

File: docs/pages/searchfield.js [hunk 2]
```diff
@@ -130,6 +130,30 @@ card(
...
+        `}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="don't"
+        title="When Not to Use"
+        description={`
+          - As a means of inputting content to a form. Use [TextField](/textfield) instead..
```
Comment (ayeshakmaz) [inline docs/pages/searchfield.js:149]: double period

File: docs/pages/segmentedcontrol.js [hunk 3]
```diff
@@ -58,6 +59,30 @@ card(
...
+          - To switch between views within a small area of content, such as a [Popover](/popover).
+        `}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="don't"
+        title="When Not to Use"
+        description={`
+          - To switch between views that represent the main content of a surface. Use [Tabs](/tabs) instead.
+          - To act as as a radio control within a form. Use [RadioButton](/radiobutton) instead.
```
Comment (ayeshakmaz) [inline docs/pages/segmentedcontrol.js:79]: double "as"

File: docs/pages/tabs.js [hunk 4]
```diff
@@ -70,6 +71,31 @@ card(
...
+          - To break up a large collection of content into logical, digestible views.
+          - To switch between different, yet related views, such as Updates and Messages.
+        `}
+      />
+      <MainSection.Card
+        cardSize="md"
+        type="don't"
+        title="When Not to Use"
+        description={`
+          - When any UI or content above the Tabs are altered upon selection. Use [Link](/link) instead.
```
Comment (ayeshakmaz) [inline docs/pages/tabs.js:91]: grammer nit: "is altered"

File: docs/pages/searchfield.js (hunk 1 above)
Comment (ponori) [reply docs/pages/searchfield.js:141]: Done!

File: docs/pages/searchfield.js (hunk 2 above)
Comment (ponori) [reply docs/pages/searchfield.js:149]: Done!

File: docs/pages/segmentedcontrol.js (hunk 3 above)
Comment (ponori) [reply docs/pages/segmentedcontrol.js:79]: Done!

File: docs/pages/tabs.js (hunk 4 above)
Comment (ponori) [reply docs/pages/tabs.js:91]: Nice! Done!

**Review by ayeshakmaz — APPROVED** [review]

**Pull request merged**